src/RMGGrabmayrGCReader.cc

```diff
@@ -0,0 +1,183 @@
+  // Skip Header
+  std::string line;
+  int header_length = 0;
+  do {
+    std::getline(file, line);
+    header_length++;
+  } while (line[0] == '%' || (line.find("version") != std::string::npos));
+
+  // In case the Random start location macro is set
+  if (fGammaCascadeRandomStartLocation) {
+    int n_entries_in_file = 0;
+    // Seems excessiv to read through the entire file, there has to be a quicker way?
+    while (std::getline(file, line)) n_entries_in_file++;
+
+    file.clear();                 // clear EOF flag
+    file.seekg(0, std::ios::beg); // move to beginning of file
+
+    int start_location = (int)(n_entries_in_file * G4UniformRand() + header_length);
+
+    RMGLog::Out(RMGLog::detail, "Random start location: ", start_location);
+    for (int j = 0; j < start_location; j++) std::getline(file, line);
+  }
+}
+
+void RMGGrabmayrGCReader::SetGammaCascadeFile(const G4int z, const G4int a, const G4String file_name) {
+
+  RMGLog::Out(RMGLog::detail, "Opening file ", file_name);
+  std::unique_ptr<std::ifstream> file = std::make_unique<std::ifstream>(file_name);
+
+  if (z == 0 || a == 0)
+    RMGLog::Out(RMGLog::fatal,
+        "Isotope Z:" + std::to_string(z) + " A: " + std::to_string(a) + " does not exist.");
+  if (!file || !file->is_open())
+    RMGLog::Out(RMGLog::fatal, "Gamma cascade file: " + file_name + " not found! Exit.");
+
+  SetStartLocation(*file);
+
+  fCascadeFiles.insert({{z, a}, std::move(file)});
+}
+
+void RMGGrabmayrGCReader::RandomizeFiles() {
+  RMGLog::Out(RMGLog::detail, "(Un)-Randomizing start locations");
+  for (auto& el : fCascadeFiles) { SetStartLocation(*el.second); }
+}
+
+void RMGGrabmayrGCReader::SetGammaCascadeRandomStartLocation(const int answer) {
+  fGammaCascadeRandomStartLocation = answer;
+  RMGLog::Out(RMGLog::detail,
+      "setting fGammaCascadeRandomStartLocation to: ", fGammaCascadeRandomStartLocation);
+  RandomizeFiles();
+}
+
+void RMGGrabmayrGCReader::DefineCommands() {
+  fGenericMessenger = std::make_unique<G4GenericMessenger>(this, "/RMG/GrabmayrGammaCascades/",
+      "Control Peters gamma cascade model");
+
+  fGenericMessenger
+      ->DeclareMethod("SetGammaCascadeRandomStartLocation",
+          &RMGGrabmayrGCReader::SetGammaCascadeRandomStartLocation)
+      .SetGuidance("Set the whether the start location in the gamma cascade file is random or not")
+      .SetGuidance("0 = don't")
+      .SetGuidance("1 = do")
+      .SetCandidates("0 1")
+      .SetDefaultValue("0")
+      .SetStates(G4State_PreInit, G4State_Idle);
+
+  // SetGammaCascadeFile cannot be defined with the G4GenericMessenger (it has to many parameters).
+  fUIMessenger = std::make_unique<GCMessenger>(this);
+}
+
+
+RMGGrabmayrGCReader::GCMessenger::GCMessenger(RMGGrabmayrGCReader* reader) : fReader(reader) {
+  fGammaFileCmd = new G4UIcommand("/RMG/GrabmayrGammaCascades/SetGammaCascadeFile", this);
+  fGammaFileCmd->SetGuidance("Set the Z, A and /path/to/file for the gamma cascade employed upon "
```

Comment:
I do not really like the message here. It seems to technical for a general description of what this command does. Maybe something akin "Set a gamma cascade file to be used upon neutron capture on a specifc isotope"?
(The parameters are already described by their own docstring, thats fine.)